crates/host/src/lib.rs

```diff
@@ -73,16 +74,29 @@ impl<'a> TryFrom<&'a str> for ResourceRef<'a> {
                         .to_file_path()
                         .map(Self::File)
                         .map_err(|()| anyhow!("failed to convert `{url}` to a file path")),
-                    "bindle" => s
-                        .strip_prefix("bindle://")
-                        .map(Self::Bindle)
-                        .context("invalid Bindle reference"),
-                    "oci" => s
-                        .strip_prefix("oci://")
-                        .map(Self::Oci)
-                        .context("invalid OCI reference"),
-                    // TODO: Support other schemes
-                    scheme => bail!("unsupported scheme `{scheme}`"),
+                    "bindle" => {
+                        // Note: bindle is not a scheme, but using this as a prefix takes out the guesswork
+                        s.strip_prefix("bindle://")
+                            .map(Self::Bindle)
+                            .context("invalid Bindle reference")
+                    }
+                    "oci" => {
+                        // Note: oci is not a scheme, but using this as a prefix takes out the guesswork
+                        s.strip_prefix("oci://")
+                            .map(Self::Oci)
+                            .context("invalid OCI reference")
+                    }
+                    scheme @ ("http" | "https") => {
```

Comment:
Can you give an example of when this code path would be taken? What's the use case of passing an OCI ref with a `http(s)` scheme?

Reply:
Sure, consider a URL like in the test: https://github.com/wasmCloud/wasmCloud/blob/ff0f52c11e3d94e3fd03ee8bc7bc211de785d453/crates/host/src/lib.rs#L237

The intent here is to gracefully process more patterns of URLs. Most references shouldn't need an `http` or `https`, but while exercising this code, I as a user tried to get URLs like `registry:5000/v2/foo:0.1.0` or `localhost:5000/v2/foo:0.1.0` to parse, including appending `http://` to the front of them